app/src/main/java/io/kuenzler/whatsappwebtogo/BlobConverter.java

```diff
@@ -0,0 +1,144 @@
+package io.kuenzler.whatsappwebtogo;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Environment;
+import android.os.Handler;
+import android.util.Base64;
+import android.webkit.JavascriptInterface;
+import android.webkit.MimeTypeMap;
+import android.widget.Toast;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.content.FileProvider;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.util.Date;
+
+public class BlobConverter {
+    private Context context;
+    public BlobConverter(Context context) {
+        this.context = context;
+    }
+
+
+    @JavascriptInterface
+    public void getBase64FromBlobData(String base64Data) throws IOException {
+        convertBase64StringToPdfAndStoreIt(base64Data);
+    }
+    public static String getBase64StringFromBlobUrl(String blobUrl) {
+        if(blobUrl.startsWith("blob")){
+            return "javascript: var xhr = new XMLHttpRequest();" +
+                    "xhr.open('GET', '"+ blobUrl +"', true);" +
+                    "xhr.responseType = 'blob';" +
+                    "xhr.onload = function(e) {" +
+                    "    if (this.status == 200) {" +
+                    "        var blobPdf = this.response;" +
+                    "        var reader = new FileReader();" +
+                    "        reader.readAsDataURL(blobPdf);" +
+                    "        reader.onloadend = function() {" +
+                    "            base64data = reader.result;" +
+                    "            Android.getBase64FromBlobData(base64data);" +
+                    "        }" +
+                    "    }" +
+                    "};" +
+                    "xhr.send();";
+
+        }
+        return "javascript: console.log('It is not a Blob URL');";
+    }
+    private void convertBase64StringToPdfAndStoreIt(String base64PDf) throws IOException {
+        final int notificationId = 1;
+        String currentDateTime = DateFormat.getDateTimeInstance().format(new Date());
+        String[] strings = base64PDf.split(",");
+        String extension;
+        switch (strings[0]) {//check image's extension
+            case "data:image/jpeg;base64":

```

Comment:
instead of a switch case we could just parse the file extension between / and ; and take that as extension, so the need to have all possible cases covered is removed (.docx and other stuff should work out of the box then)